plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/exporter/QueryInsightsExporter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.exporter;
+
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+
+import java.util.List;
+
+/**
+ * Simple abstract class to export data collected by search query analyzers
+ * <p>
+ * Mainly for use within the Query Insight framework
+ *
+ * @opensearch.internal
+ */
+public abstract class QueryInsightsExporter<T extends SearchQueryRecord<?>> {
+    private QueryInsightsExporterType type;
+    private String identifier;
```

Comment:
Can these be made `final`? 

If we want to export to a different sink, we can create a new exporter.

Also, do we need a `QueryInsightsExporterType` field here? Do the types just correspond to the subclasses? If so, couldn't the subclass return its own type?